components/micro-gateway-core/src/main/ballerina/src/gateway/utils/observability_util.bal

```diff
@@ -0,0 +1,158 @@
+// Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/http;
+import ballerina/log;
+import ballerina/observe;
+import ballerina/runtime;
+
+boolean isTracingEnabled = getConfigBooleanValue(MICRO_GATEWAY_TRACING, ENABLED, false);
+boolean isMetricsEnabled = getConfigBooleanValue(MICRO_GATEWAY_METRICS, ENABLED, false);
+
+//metrics related methods
+public function initializeGauge(string name, string description, map<string> | () gaugeTags) returns observe:Gauge | () {
+    if (isMetricsEnabled == false) {
+        return ();
+    } else {
+        observe:Gauge localGauge = new (name, description, gaugeTags);
+        registerGauge(localGauge);
+        return localGauge;   
+    }
+}
+
+public function setGaugeDuration(int starting) returns float | () {
+    if (isMetricsEnabled == false) {
+        return ();
+    }
+    else {
+        int ending = getCurrentTime();
+        float latency = (ending - starting) * 1.0;
+        return (latency);
+    }
+}
+
+public function updateGauge(observe:Gauge | () localGauge, float | () latency) {
+    if (isMetricsEnabled) {
+        observe:Gauge gauge = <observe:Gauge>localGauge;
+        float gauge_latency = <float>latency;
+        gauge.setValue(gauge_latency);
+    }
+}
+
+public function registerGauge(observe:Gauge gauge) {
+    error? result = gauge.register();
+    if (result is error) {
+        log:printError("Error in registering Counter", err = result);
+    }
+}
+
+public function gaugeTagDetails(http:Request request, http:FilterContext context, string category) returns map<string> | () {
+    if (isMetricsEnabled == false) {
+        return ();
+    }
+    else {
+        map<string> gaugeTags = {"Category": category, "Method": request.method, "ServicePath": request.rawPath, "Service": context.getServiceName()};
+        return gaugeTags;   
+    }
+}
+
+public function gaugeTagDetails_authn(http:Request request, string category) returns map<string> | () {
+    if (isMetricsEnabled == false) {
+        return ();
+    }
+    else {
```

Comment:
```diff
- }
- else {
+ } else {
```
This has to be corrected throughout this file. I'll send a different PR with the fix since @ShanJayaweera is no longer available.

Reply:
#907